File: src/main/java/HOCONParser.java
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import com.typesafe.config.ConfigValueType;

import java.io.File;
import java.util.HashSet;
import java.util.Set;

public class HOCONParser {

    public static Set<String> parseConfig(String filePath) {
        Config config = ConfigFactory.parseFile(new File(filePath));
        return getAllKeys(config);
    }

    private static Set<String> getAllKeys(Config config) {
        return getAllKeys(config, "");
    }

    private static Set<String> getAllKeys(Config config, String parentKey) {
        Set<String> keys = new HashSet<>();
        for (String key : config.root().keySet()) {
            String fullKey = parentKey.isEmpty() ? key : parentKey + "." + key;
            if (config.getValue(key).valueType() == ConfigValueType.OBJECT) {
                keys.addAll(getAllKeys(config.getConfig(key), fullKey));
            } else {
                keys.add(fullKey);
            }
        }
        return keys;
    }
}

File: src/main/scala/HOCONParser.scala
package com.example

import com.typesafe.config.{Config, ConfigFactory}

import scala.collection.JavaConverters._

object HOCONParser {
  def parseConfig(filePath: String): Set[String] = {
    val config: Config = ConfigFactory.parseFile(new java.io.File(filePath))
    getAllKeys(config).toSet
  }

  private def getAllKeys(config: Config, parentKey: String = ""): Set[String] = {
    config.entrySet().asScala.flatMap { entry =>
      val key = if (parentKey.isEmpty) entry.getKey else s"$parentKey.${entry.getKey}"
      entry.getValue.valueType() match {
        case com.typesafe.config.ConfigValueType.OBJECT =>
          getAllKeys(config.getConfig(entry.getKey), key)
        case _ =>
          Set(key)
      }
    }.toSet
  }

  def getConfigValue(filePath: String, key: String): String = {
    val config: Config = ConfigFactory.parseFile(new java.io.File(filePath))
    config.getString(key)
  }
}
